lib/sensors/temperature.cpp

```diff
@@ -0,0 +1,53 @@
+#include "temperature.hpp"
+
+namespace hyped::sensors {
+
+Temperature::Temperature(hyped::io::I2c &i2c, hyped::core::ILogger &log) : log_(log), i2c_(i2c)
+{
+}
+
+std::optional<std::uint16_t> Temperature::read()
+{
+  const auto Status_check = i2c_.readByte(kMux, kStatus);
```

Comment:
i think just change the left to `const std::optional<std::uint8_t> status_check_result`